core/players-handbook-2024/classes/class-ranger.xml

```diff
@@ -0,0 +1,419 @@
+			<stat name="favoredenemy:usage" value="1" level="17"/>
+		</rules>
+	</element>
+	<element name="Level 1: Weapon Mastery" type="Class Feature" source="Player’s Handbook 2024" id="ID_WOTC_PHB24_CLASS_FEATURE_RANGER_WEAPON_MASTERY">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RANGER_24_WEAPON_MASTERY</requirements>
+		<description>
+			<p>Your training with weapons allows you to use the mastery properties of two kinds of weapons of your choice with which you have proficiency, such as Longbows and Shortswords.</p>
+			<p class="indent">Whenever you finish a Long Rest, you can change the kinds of weapons you chose. For example, you could switch to using the mastery properties of Scimitars and Longswords.</p>
+		</description>
+		<sheet alt="Weapon Mastery">
+			<description>You can use the mastery properties of two kinds of weapons of your choice with which you have proficiency. Whenever you finish a Long Rest, you can change the kinds of weapons you chose.</description>
+		</sheet>
+		<rules>
+			<select type="Class Feature" name="Weapon Mastery" supports="Weapon Mastery" number="2"/>
+		</rules>
+	</element>
+	<element name="Level 2: Deft Explorer" type="Class Feature" source="Player’s Handbook 2024" id="ID_WOTC_PHB24_CLASS_FEATURE_RANGER_DEFT_EXPLORER">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RANGER_24_DEFT_EXPLORER</requirements>
+		<description>
+			<p>Thanks to your travels, you gain the following benefits.</p>
+			<p class="indent"><i>Expertise.</i> Choose one of your skill proficiencies with which you lack Expertise. You gain Expertise in that skill.</p>
+			<p class="indent"><i>Languages.</i> You know two languages of your choice from the language tables in chapter 2.</p>
+		</description>
+		<sheet display="false"/>
+		<rules>
+			<select type="Proficiency" name="Expertise (Deft Explorer)" supports="ID_INTERNAL_SUPPORT_EXPERTISE"/>
+			<select type="Language" name="Language (Deft Explorer)" supports="Standard||Exotic" number="2" />
+		</rules>
+	</element>
+	<element name="Level 2: Fighting Style" type="Class Feature" source="Player’s Handbook 2024" id="ID_WOTC_PHB24_CLASS_FEATURE_RANGER_FIGHTING_STYLE">
+		<requirements>!ID_INTERNAL_FEATURE_REPLACEMENT_RANGER_24_FIGHTING_STYLE</requirements>
+		<description>
+			<p>You gain a Fighting Style feat of your choice (see chapter 5). Instead of choosing one of those feats, you can choose the option below.</p>
+			<div class="reference">
+				<div element="ID_WOTC_PHB24_FEAT_DRUIDIC_WARRIOR" />
+			</div>
+		</description>
+		<sheet display="false"/>
+		<rules>
+			<select type="Feat" name="Fighting Style" supports="Fighting Style" />
+		</rules>
+	</element>
+	<!-- Druidic Warrior FEAT -->
+	<element name="Druidic Warrior" type="Feat" source="Player’s Handbook 2024" id="ID_WOTC_PHB24_FEAT_DRUIDIC_WARRIOR">
+		<requirements>ID_WOTC_PHB24_CLASS_FEATURE_RANGER_FIGHTING_STYLE</requirements>
+		<supports>Fighting Style</supports>
+		<description>
```

Comment:
Currently the app shows no "flavor text" in the select menu, unlike the other fighting styles. This should fix that.
```suggestion
		<prerequisite>Ranger Fighting Style Feature, Fighting Style Feat</prerequisite>
		<description>
```